pkg/notifiers/sender.go

```diff
@@ -325,6 +343,25 @@ func smtpInit(host string, port int) (*smtp.Client, error) {
 	return c, nil
 }
 
+func dialerWithTLSConfig(dialer dialer.Dialer, host, smartHost string) (dialer.Dialer, error) {
```

Comment:
The error return value of the function looks unnecessary because it is always nil

Reply:
oh, https://github.com/rancher/rancher/blob/37ecfd80d08309863d7d414163a1b2271cbab750/pkg/notifiers/sender.go#L354
and L359 return error?

Reply:
Those are errors returned from `dialer.Dialer`, I mean the return value of the  `dialerWithTLSConfig` function